### PR TITLE
Add wizard to configure db-sync

### LINKED ISSUE #499: Wizard for database sync
It would be good to have a simple wizard to create initial DB sync configuration file.

There would be a new menu entry in Plugins menu > Mergin Maps, that would open the wizard. This would only work when a MM project is currently opened (if not, it would suggest to user to open a MM project first).

Overall workflow:

mergin-db-sync-config-wizard

Source excalidraw file: [mergin-db-sync-config-wizard.excalidraw.zip](https://github.com/MerginMaps/qgis-mergin-plugin/files/11714758/mergin-db-sync-config-wizard.excalidraw.zip)

### PR DESCRIPTION
**Pull request by alexbruy**
Add a simple wizard to create initial config for `db-sync` for currently opened MerginMaps project.
wizard
Fixes #499

### CODE REVIEW

**Comment by jozef-budac** [comment]
For the configuration to somehow work, I had to update the configuration this way
1) sync_file: had an absolute path to the gpkg, but it needs to contain only the name of the gpkg
2) conn_info: it has only host and dbname information. user and password were missing and I had to add them

with this updates I get this output. There is an error on the end, but it is probably another problem
```
jozef@ThinkPad-X1:~/Lutra/mergin-db-sync$ docker run -it --network="host" -v ${PWD}:/config lutraconsulting/mergin-db-sync_newversion:latest /config/config.yaml
== starting mergin-db-sync daemon == version 2.1.0 ==
Using config file: /config/config.yaml
Email notifications for sync failures are not set up. It is recommended to use them. Please see documentation.
Logging in to Mergin...
Going to log in user tin1
User tin1 successfully logged in.
Processing Mergin Maps project 'tin1/judasda'
Connecting to the database...
Downloading latest Mergin Maps project tin1/judasda to /tmp/dbsync/judasda
--- version: Python-client/0.8.3 DB-sync/2.1.0 (Linux/Ubuntu)
--- start download tin1/judasda
got project info. version v1
will download 2 files in 2 chunks, total size 115061
Downloading Survey_points.gpkg version=v1 diff=False part=0
Downloading judasda.qgz version=v1 diff=False part=0
Download finished: Survey_points.gpkg
Download finished: judasda.qgz
--- download finished
The base and modified schemas do not exist yet, going to initialize them ...
GEODIFF: NOTICE:  schema "newsync" does not exist, skipping

GEODIFF: NOTICE:  schema "newsync_db_sync" does not exist, skipping

Init done!
Traceback (most recent call last):
  File "dbsync_daemon.py", line 198, in <module>
    main()
  File "dbsync_daemon.py", line 166, in main
    config.notification.minimal_email_interval if "minimal_email_interval" in config.notification else 4
  File "/usr/local/lib/python3.8/dist-packages/dynaconf/base.py", line 145, in __getattr__
    value = getattr(self._wrapped, name)
  File "/usr/local/lib/python3.8/dist-packages/dynaconf/base.py", line 326, in __getattribute__
    return super().__getattribute__(name)
AttributeError: 'Settings' object has no attribute 'NOTIFICATION'
```

**Comment by jozef-budac** [comment]
Small update. the error on the end is a problem in dbsync_daemon.py on line 166. Martin D knows about it and will probably fix it.

**Comment by alexbruy** [comment]
@jozef-budac can you provide mode details on how your postgres connection is configured in QGIS? Are you using authid or plain text login/password?

**Comment by jozef-budac** [comment]
@alexbruy I checked the connection settings in QGIS and I saw that the password and username were not stored there. After I updated the credentials, the generated configuration contained both the user and password for the Postgres connection.

**Comment by alexbruy** [comment]
@jozef-budac thanks for the update!

File: Mergin/ui/ui_config_file_page.ui [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        </size>
+       </property>
+      </spacer>
+     </item>
+    </layout>
+   </item>
+   <item>
+    <widget class="QLabel" name="label_2">
+     <property name="text">
+      <string>See documentation on how to run the sync tool with generated configuration file</string>
```
Comment (PeterPetrik) [inline Mergin/ui/ui_config_file_page.ui:71]: use URL https://merginmaps.com/docs/dev/dbsync/

File: Mergin/ui/ui_sync_direction_page.ui [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    <height>174</height>
+   </rect>
+  </property>
+  <property name="windowTitle">
+   <string>WizardPage</string>
+  </property>
+  <layout class="QVBoxLayout" name="verticalLayout">
+   <item>
+    <widget class="QLabel" name="label">
+     <property name="text">
```
Comment (PeterPetrik) [inline Mergin/ui/ui_sync_direction_page.ui:19]: Can you add a section in the first wizard page with link to https://merginmaps.com/docs/dev/dbsync/
and a short description of tool like

"Mergin Maps DB Sync is a tool that takes care of two-way synchronisation between Mergin Maps and PostGIS databases. Read the description on URL"

File: Mergin/utils.py [hunk 3]
```diff
@@ -797,11 +797,7 @@ def update_datasource(layer, new_path):
     """Updates layer datasource, so the layer is loaded from the new location"""
     options = QgsDataProvider.ProviderOptions()
     options.layerName = layer.name()
-    if layer.dataProvider().name() == "mbtilesvectortiles":
-        # For 3.31 and master
-        layer.setDataSource(f"url={new_path}&type=mbtiles", layer.name(), layer.dataProvider().name(), options)
-    elif layer.dataProvider().name() == "vectortile":
-        # For 3.22
```
[the remark this answers is not in the document]
Comment (alexbruy) [reply Mergin/utils.py:804]: Because it is already in master

File: Mergin/ui/ui_sync_direction_page.ui (hunk 2 above)
Comment (alexbruy) [reply Mergin/ui/ui_sync_direction_page.ui:19]: Done
image

File: Mergin/ui/ui_config_file_page.ui (hunk 1 above)
Comment (alexbruy) [reply Mergin/ui/ui_config_file_page.ui]: Done.

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**